File: 01-12-2024/A. New Year and Hurry.cpp
/*
    Author: Tajwar Saiyeed
    Date: 2024-12-02 22:23:16
    File: A. New Year and Hurry.cpp
*/


#include <iostream>
using namespace std;


int main()
{
    int n, k;
    cin >> n >> k;

    int totalMinutes = 240;
    int availableMinutes = totalMinutes - k;
    int timeSpent = 0, solvedProblems = 0;

    for (int i = 1; i <= n; ++i)
    {
        timeSpent += 5 * i;
        if (timeSpent > availableMinutes)
        {
            break;
        }
        ++solvedProblems;
    }

    cout << solvedProblems << endl;
    return 0;
}

File: 05-12-2024/A. New Year and Hurry.py
n, k = map(int, input().split())

total_minutes = 240
available_minutes = total_minutes - k
time_spent = 0
solved_problems = 0

for i in range(1, n + 1):
    time_spent += 5 * i
    if time_spent > available_minutes:
        break
    solved_problems += 1

print(solved_problems)